sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/Transforms.java

```diff
@@ -381,12 +381,22 @@ static PathItem toPathItem(Path path) {
             parseDateOrNull(path.getLastModified()), path.getContentLength() == null ? 0 : path.getContentLength(),
             path.getGroup(), path.isDirectory() != null && path.isDirectory(), path.getName(), path.getOwner(),
             path.getPermissions(),
-            path.getCreationTime() == null ? null : fromWindowsFileTimeOrNull(Long.parseLong(path.getCreationTime())),
-            path.getExpiryTime() == null ? null : fromWindowsFileTimeOrNull(Long.parseLong(path.getExpiryTime())));
+            parseWindowsFileTimeOrDateString(path.getCreationTime()),
+            parseWindowsFileTimeOrDateString(path.getExpiryTime()));
 
         return AccessorUtility.getPathItemAccessor().setPathItemProperties(pathItem, path.getEncryptionScope(), path.getEncryptionContext());
     }
 
+    private static OffsetDateTime parseWindowsFileTimeOrDateString(String date) {
```

Comment:
Let's make this `static` and add a test specifically for this that isn't recorded where we pass

- A Windows time as a long
- Date Time RFC1123 string
- Some junk
- null

And make sure we get what we expect.